tests/integration_tests/build/test_coverage.py

```diff
@@ -30,6 +30,12 @@
 KCOV_COVERAGE_REGEX = r'"covered":"(\d+\.\d)"'
 """Regex for extracting coverage data from a kcov output file."""
 
+KCOV_COVERED_LINES_REGEX = r'"covered_lines":"(\d+)"'
```

Comment:
Is there any possibility to actually get the kcov html output, zip it and expose it somehow, or otherwise just expose the kcov output folder? I can actually take a more educated look at that failure and also it would hint me places were I need to add new tests.

Reply:
This PR mostly wants to give you the possibility to avoid situation where after a rebase your PR fails coverage even though before rebase it used to pass. It would let you know when your coverage is at risk of falling under the limit. It is not meant to help you detect lines that need additional coverage.
Regardless of that, I still think that this is a nice idea that can be added to #1181 .

Reply:
Integration with coveralls.io or something similar would be great.

Reply:
@petreeftime coveralls.io doesn't support bare metal instances as far as I know, so we won't be able to use it.

Reply:
I believe `cargo kcov` can produce an archive that can be sent to coveralls.io via an API, so if the baremetal machine has Internet access it should be possible to use it, even if you don't use Travis CI or some other build system that can be directly integrated into Github. It's worth checking out, at any rate.